src/shared/schemas.ts

```diff
@@ -109,7 +109,8 @@ export class SchemaService {
         const isSchemaChanged =
             this.owned.has(mapping.uri) &&
             mapping.owner !== undefined &&
-            this.owned.get(mapping.uri)?.schema !== mapping.schema
```

Comment:
The PR description mentions something about not having an owner, but these conditions are already checking that.

There are subtleties here. So, a 1-line comment is important to explain the subtlety.